ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -204,8 +222,124 @@ private String genBlobName(String fileName, String databaseName, String tableNam
         return String.format("%s__%s__%s__%s", databaseName, tableName, UUID.randomUUID().toString(), fileName);
     }
 
+    /**
+     * Ingests a ResultSet into the service
+     *
+     * @param resultSetSourceInfo The specific SourceInfo to be ingested.
+     * @param ingestionProperties Settings used to customize the ingestion operation
+     * @return A result that could be used to query for status
+     * @throws IngestionClientException  An exception originating from a client activity
+     * @throws IngestionServiceException An exception returned from the service
+     */
     @Override
-    public IngestionResult ingestFromResultSet(ResultSetSourceInfo resultSetSourceInfo, IngestionProperties ingestionProperties) {
-        throw new UnsupportedOperationException();
+    public IngestionResult ingestFromResultSet(ResultSetSourceInfo resultSetSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        try {
+            Objects.requireNonNull(resultSetSourceInfo, "resultSetSourceInfo cannot be null");
+            resultSetSourceInfo.validate();
+
+            File tempFile = File.createTempFile("kusto-resultset", ".csv.gz");
```

Comment:
Since we don't know the size of the resultset we can't know if we'll run out of space.
I've added an overload that enables the user to specify the temp path.